js/source/source_cache.js

```diff
@@ -255,7 +262,15 @@ class SourceCache extends Evented {
             }
             if (this._cache.has(coord.id)) {
                 retain[coord.id] = true;
-                return this._cache.get(coord.id);
+                const cachedTile = this._cache.get(coord.id);
+
+                if (this._cacheTimers[coord.id]) {
+                    clearTimeout(this._cacheTimers[coord.id]);
+                    delete this._cacheTimers[coord.id];
+                    this._setTimers(coord.id, cachedTile);
+                }
```

Comment:
It looks to me like we always do `addTile()` with nonempty return value from this method -- since that's the case, and since we have equivalent logic for removing cacheTimer / adding timer already in addTile, is it possible that we could avoid redundancy by only doing it in addTile?